### DISCUSSION TITLE
Repology is not fetching right versions

### DISCUSSION THREAD

**Discussion by okainov** [renovatebot/renovate#26619] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest docker

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

We have a few dependencies where we look up the versions from Repology. One of the recent ones was `ncurses`, we used `# renovate: datasource=repology depName=conancenter/ncurses`. However it stopped working

I thought the reason might be in conancenter itself, so I tried different depname from `fedora_rawhide`, but still getting nothing.

From the logs:

```
 GET https://repology.org/tools/project-by?repo=fedora_rawhide&name_type=binname&target_page=api_v1_project&noautoresolve=on&name=ncurses = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=379) (repository=myrepo)
```

And indeed this URL doesn't really return anything useful:  https://repology.org/tools/project-by?repo=fedora_rawhide&name_type=binname&target_page=api_v1_project&noautoresolve=on&name=ncurses:

> This link was supposed to redirect you to Repology project corresponding to binary package name ncurses in repository identified as fedora_rawhide, but it has failed because the specified package name was not found.

But I see the versions in general list of the packages https://repology.org/project/ncurses/versions... 

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by cevich** [comment]
I'm having this problem as well, but with the `fedora_39` repo.  Messing around, I did find if you change the (broken) URL renovate tries to use, `s/name_type=binname/name_type=srcname`, I get back a JSON result.

Doing this [with your example](https://repology.org/tools/project-by?repo=fedora_rawhide&name_type=srcname&target_page=api_v1_project&noautoresolve=on&name=ncurses), I get a giant JSON map list including `ncurses` from every repo.  One of the list items is:

```json
...snip...
  {
    "repo": "fedora_rawhide",
    "subrepo": "development",
    "srcname": "ncurses",
    "visiblename": "ncurses",
    "version": "6.4",
    "licenses": [
      "MIT-open-group"
    ],
    "summary": "Ncurses support utilities",
    "categories": [
      "Unspecified"
    ],
    "status": "newest",
    "origversion": "6.4-9.20240113.fc40"
  },
...snip...
```

So I think maybe there are two problems here:
1. Renovate is using the wrong repology API or using it incorrectly
2. The repology setup for the Fedora repos is wrong, since no `binname` type searches work (I tried several).

**Reply by cevich** [reply]
Update: I tried plugging in the `alpine_3_12/gcc` [example from the docs](https://docs.renovatebot.com/modules/datasource/repology/).  That one does return JSON, with `name_type=binname`.  So I even more strongly believe there's some kind of misconfiguration on the repology-side with the Fedora repos. (I haven't tried others).

**Reply by okainov** [reply]
>Renovate is using the wrong repology API or using it incorrectly

I am actually more in favor of this, because I'd expect Renovate to check for the versions which I see on the website, regardless of whether they're considered "sources" or "binaries"... Maybe there can be some extra option to tell it "only look up for binaries", but just doing this out of the box seems wrong for me. 

At least we're using it as *pure* version indicator, the actual libs are then fetched from somewhere else, not from Repology...

**Reply by cevich** [reply]
Yeah, having a source or binary selector would be nice.  Though I suspect 99.9999% users want it to be 'binary'.  I noticed in the listings, there's no "maintainer" listed for any of the Fedora repos.  Perhaps that's the problem?  Repology is guessing at how to support Fedora repos and getting it wrong.

The other thing I noticed w/ the versions I could look up: None of the versions include the release component.  e.g. on the `podman` package, repology shows a version like "4.8.3" but not "4.8.3-1.fc39" which is what `dnf` considers the "version".  Though it's the same behavior for CentOS Stream 9, so maybe this is intentional.  Anyway, it doesn't matter since none of the Fedora repos work with the repology manager anyway :disappointed: 

I have a (somewhat) minimal reproducer for this if that helps fix the problem.  But I suspect more and more the problem is between repology <--> Fedora.
